portal/src/main/webapp/js/src/dashboard/iviz.js

```diff
@@ -8178,3 +8179,191 @@ window.LogRankTest = (function(jStat) {
+                        'iviz-share-cohort-btn-qtip'
+                    },
+                    show: {event: 'click', ready: false},
+                    hide: false,
+                    position: {
+                        my: 'top center',
+                        at: 'bottom center',
+                        viewport: $(window)
+                    },
+                    events: {
+                        render: function(event, api) {
+                            var tooltip = $('.iviz-share-cohort-btn-qtip .qtip-content');
+                            
+                            tooltip.find('.share-cohort').click(function() {
+                                tooltip.find('.shared').css('display', 'none');
+                                tooltip.find('.dialog').css('display', 'none');
+                                api.reposition();
+                                
+                                // Copy virtual study link to clipboard
+                                var $temp = $("<input>");
+                                $("body").append($temp);
+                                $temp.val(tooltip.find('.virtual-study-link').attr('href')).select();
+                                document.execCommand("copy");
+                                // Check if users copy url successfully
+                                if ($temp.val() === tooltip.find('.virtual-study-link').attr('href')) {
+                                    tooltip.find('.shared').css('display', 'block');
+                                    tooltip.find('.dialog').css('display', 'none');
+                                    $temp.remove();
+                                    api.reposition();
+                                }
+                            });
+                            
+                        },
+                        show: function() {
+                            var tooltip = $('.iviz-share-cohort-btn-qtip .qtip-content');
+                            tooltip.find('.dialog').css('display', 'block');
+                            tooltip.find('.shared').css('display', 'none');
+                            
+                            var cohortName = $('#study_name').val();
+                            var cohortDescription = $('#study_desc').val();
+                            if (_.isObject(vcSession)) {
+                                self_.updateStats = true;
+                                
+                                self_.$nextTick(function() {
+                                    var _selectedSamplesNum = 0;
+                                    var _selectedPatientsNum = 0;
+                                    var saveCohort = false;
+                                    
+                                    if (_.isObject(self_.stats.selectedCases)) {
+                                        var different_patients = [];
+                                        var different_samples = [];
```

Comment:
I dont think you need to define these (`different_patients`, `different_samples`) up here, you only use them once